chain/substreams/src/data_source.rs

```diff
@@ -48,8 +46,16 @@ impl blockchain::DataSource<Chain> for DataSource {
     }
 
     fn start_block(&self) -> BlockNumber {
-        // TODO(filipe): Figure out how to handle this
-        0
+        match self.source.package.modules {
+            Some(ref modules) => modules
+                .modules
+                .iter()
+                .min_by(|x, y| x.initial_block.cmp(&y.initial_block))
+                .map(|m| m.initial_block.try_into().map_err(anyhow::Error::from))
+                .unwrap_or(Ok(0))
```

Comment:
Not great that errors are ignored here, perhaps this start block value should be extracted and stored when the data source is resolved?

Reply:
Yeah it's a good idea